vernac/g_vernac.mlg

```diff
@@ -744,18 +744,19 @@ GRAMMAR EXTEND Gram
           { VernacCanonical CAst.(make ~loc @@ ByNotation ntn) }
 
       (* Coercions *)
-      | IDENT "Coercion"; qid = global; u = OPT univ_decl; d = def_body ->
-          { let s = coerce_reference_to_id qid in
-          VernacDefinition ((NoDischarge,Coercion),((CAst.make ?loc:qid.CAst.loc (Name s)),u),d) }
+      | IDENT "Coercion"; qid = global; ud = OPT [ u = OPT univ_decl; d = def_body -> { u, d } ] ->
+          { match ud with Some (u, d) -> let s = coerce_reference_to_id qid in
+          VernacDefinition ((NoDischarge,Coercion),((CAst.make ?loc:qid.CAst.loc (Name s)),u),d)
+          | None -> VernacCoercion (CAst.make ~loc @@ AN qid, None) }
       | IDENT "Identity"; IDENT "Coercion"; f = identref; ":";
          s = class_rawexpr; ">->"; t = class_rawexpr ->
            { VernacIdentityCoercion (f, s, t) }
       | IDENT "Coercion"; qid = global; ":"; s = class_rawexpr; ">->";
          t = class_rawexpr ->
-          { VernacCoercion (CAst.make ~loc @@ AN qid, s, t) }
+          { VernacCoercion (CAst.make ~loc @@ AN qid, Some(s, t)) }
       | IDENT "Coercion"; ntn = by_notation; ":"; s = class_rawexpr; ">->";
          t = class_rawexpr ->
-          { VernacCoercion (CAst.make ~loc @@ ByNotation ntn, s, t) }
+          { VernacCoercion (CAst.make ~loc @@ ByNotation ntn, Some (s, t)) }
```

Comment:
You didn't change the grammar here.  Did you mean to?

Reply:
Yes and no, c.f. above comment.